sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/AvroGenericRecordToStorageApiProto.java

```diff
@@ -115,34 +120,97 @@ static String convertUUID(Object value) {
   }
 
   static Long convertTimestamp(Object value, boolean micros) {
-    if (value instanceof ReadableInstant) {
-      return ((ReadableInstant) value).getMillis() * (micros ? 1000 : 1);
+    if (value instanceof org.joda.time.ReadableInstant) {
```

Comment:
may worth add a comment noting the following helper functions support both joda time and java.time, and refer to #19215 as future clean up

Reply:
! don't think this is related. `joda-time` must be supported here as long as avro `1.8` and `1.9` are supported by beam [here](https://github.com/apache/beam/blob/3d651eb51ddda8ef71f190de39a4e1e43b1c20ea/sdks/java/extensions/avro/build.gradle#L40).